### ISSUE TITLE
add tags to challenge POST 

### ISSUE THREAD

**Issue by JMurphyWeb** [emfoundation/ce100-app#439]
add tags to challenge POST endpoint

endpoint: /challenges/id/tags (POST)

Very much linked to add tags to org post endpoint https://github.com/emfoundation/ce100-app/issues/430

**Comment by JMurphyWeb** [comment]
Sadly the permissions are not as easy check as they were in orgs as we store the users org id in the session we can quickly check permissions in the handler:

```
  if (loggedIn.organisation_id !== challengeId && loggedIn.scope !== 'admin') {
    return reply(Boom.unauthorized('You do not have permission to edit that organisation.'));
  }

```

This is not possible in challenges so will either mean 2 queries in 1 handler (first to check permissions) or to write a new query only for challenges.

Any suggestions @SimonLab or @mk4111 ?

**Comment by SimonLab** [comment]
@JMurphyWeb sorry can we recap this endpoint just to check that I didn't missed anything?

Is the endpoint is a POST on /challenges/{idChall}/tags or /challenges/tags? It seems for the organisations the [issue  430d escription](https://github.com/emfoundation/ce100-app/issues/430) use /orgs/tags with post but on the [epic issue 366](https://github.com/emfoundation/ce100-app/issues/366) it seems that we use /orgs/{idOrgs}/tags. And I think the format will be the same for challenges, we just need to be sure we have the right one. I think it should be /challenges/{idChall}/tags right?

Are the permission for adding tags to a challenge the same as the orgs endpoint:

Permissions:
Only an admin or a primatry user that is linked with the org where the challenge is linked to.

The the issue is to be able to know if the challenge we want to add tags to belongs to the organisation of the primary user connected, is this right?

I think I would prefer at the moment to create another request in the handler which will tell us if the user logged in as the right to add a tag to a specific challenge instead of creating a specific query in the plugin. We might need to find a way to group the "permission query" together. I think having two separate requests will be quicker to integrate the feature.

Let's try to implement with the quickest and easier way but if you really prefer to include the permission feature in the plugin by creating a specific query in the plugin itself and if you think it can be done easily I don't mind to go this way also.

**Comment by JMurphyWeb** [comment]
@SimonLab you are right. I mistyped the endpoint in #430 description. It should have been: 

endpoint: /orgs/id/tags (POST)

**Comment by JMurphyWeb** [comment]
@SimonLab thanks for your input, sorry this issue is not up to scratch. I have had some pain in my wrists last 2 days so keeping typing to a minimum...

**Comment by SimonLab** [comment]
resolved with #440
